xaitk_saliency/impls/gen_image_classifier_blackbox_sal/rise.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any, Dict, Optional
+
+import numpy as np
+from smqtk_classifier import ClassifyImage
+
+from xaitk_saliency.impls.perturb_image.rise import RISEGrid
+from xaitk_saliency.impls.gen_image_classifier_blackbox_sal.occlusion_based import PerturbationOcclusion
+from xaitk_saliency.impls.gen_classifier_conf_sal.rise_scoring import RISEScoring
+from xaitk_saliency.interfaces.gen_image_classifier_blackbox_sal import GenerateImageClassifierBlackboxSaliency
+
+
+class RISEStack (GenerateImageClassifierBlackboxSaliency):
+    """
+    Encapsulation of the perturbation-occlusion method using specifically the
+    RISE implementations of the component algorithms.
+
+    This more specifically encapsulates the original RISE method as presented
+    in their paper and code. See references in the :class:`RISEGrid`
+    and :class:`RISEScoring` documentation.
+
+    This implementation shares the `p1` probability with the internal
+    `RISEScoring` instance use, effectively causing this implementation to
+    utilize debiased RISE.
+
+    :param n:
+        Number of random masks used in the algorithm. E.g. 1000.
+    :param s:
+        Spatial resolution of the small masking grid. E.g. 8.
+        Assumes square grid.
+    :param p1:
+        Probability of the grid cell being set to 1 (otherwise 0).
+        This should be a float value in the [0, 1] range. E.g. 0.5.
+    :param seed:
+        A seed to pass into the constructed random number generator to allow
+        for reproducibility
+    :param threads: The number of threads to utilize when generating masks.
+        If this is <=0 or None, no threading is used and processing
+        is performed in-line serially.
+    """
+
+    def __init__(
+        self,
+        n: int,
+        s: int,
+        p1: float,
+        seed: Optional[int] = None,
+        threads: int = 0
+    ):
+        self._po = PerturbationOcclusion(
+            RISEGrid(n=n, s=s, p1=p1, seed=seed, threads=threads),
+            RISEScoring(p1=p1),
```

Comment:
Currently, this provides the de-biased form of RISE, i.e. p1 is coupled between the `RISEGrid` and `RISEScoring` methods. Should we add a flag in the constructor for something like `debiased = True`? This could allow `p1` to be set to 0, which is for standard RISE.

Reply:
I thought about it, but decided to keep it simple to start and see if there was any commentary on it! :) I can make it a boolean enable flag. I was thinking to default it to `True` unless you think otherwise.